test/NuGet.Core.Tests/NuGet.Versioning.Test/SemanticVersionTests.cs

```diff
@@ -109,6 +109,20 @@ public void ToString_ClassExtendingSemanticVersion_ReturnsDefaultFormat()
             Assert.Equal("1.2.3", result);
         }
 
+        [Fact]
+        public void Metadata_NoNullableWarning_After_HasMetadata_checked()
+        {
+            // Arrange
+            SemanticVersion target = new(1, 2, 3);
+
+            // Act
+            // should not result in a compiler warning CS8602: Dereference of a possibly null reference.
+            string result = target.HasMetadata ? target.Metadata[0..1] : "no-metadata";
```

Comment:
This is failing with error

> test\NuGet.Core.Tests\NuGet.Versioning.Test\SemanticVersionTests.cs(120,66): Error CS0518: Predefined type 'System.Range' is not defined or imported

As `System.Range` doesn't ship with .NET Framework.

Frankly, the non-range syntax seems much simpler to understand when pulling out a single element anyway.

```suggestion
            string result = target.HasMetadata ? target.Metadata[0] : "no-metadata";
```

Reply:
The point was to access a member of the `string` type, but I'd rather use Substring then, because indexer returns `char` and any conversions shouldn't be a part of this test. So I'll change as follows:

```suggestion
            string result = target.HasMetadata ? target.Metadata.Substring(1) : "no-metadata";
```